### PR TITLE
Test that value returned content is always consumable only once

### LINKED ISSUE #204: Publisher produced by `Storage.value` method should be readable once
`S3Storage` and `InMemoryStorage` should return publishers that behave in same way. Now `S3Storage` produces publishers readable once and publisher from `InMemoryStorage` could be read multiple times.

### PR DESCRIPTION
**Pull request by Sammers21**
Closes #204 

* A test which ensures that the returned content is always consumable only once has been created
* FileStorage was fixed so the test passes

### CODE REVIEW

**Comment by Sammers21** [comment]
@olegmoz, review the Pull Request please

File: src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java [hunk 1]
```diff
@@ -115,6 +115,24 @@ void shouldFailToSaveErrorContent(final Storage storage) throws Exception {
...
+            IllegalStateException.class,
+            () -> {
+                final Key.From key = new Key.From("key");
+                storage.save(
+                    key,
+                    new Content.OneTime(new Content.From("val".getBytes()))
+                ).join();
+                final Content value = storage.value(key).join();
+                Flowable.fromPublisher(value).toList().blockingGet();
+                Flowable.fromPublisher(value).toList().blockingGet();
```
Comment (olegmoz) [inline src/test/java/com/artipie/asto/StorageSaveAndLoadTest.java:131]: @Sammers21 please minimize this failing section to just the last line, so we are sure that the code fails exactly where we expect it to fail

**Comment by 0crat** [comment]
This pull request [#210](https://github.com/artipie/asto/issues/210) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/131fcc77-afda-4b27-835e-dc9824e00223); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Job [#210](https://github.com/artipie/asto/issues/210) is closed, can't start order

**Comment by 0crat** [comment]
Job was finished in 1 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @vatavuk[/z](https://www.0crat.com/u/vatavuk), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by g4s8** [comment]
@0crat quality bad

**Pull request merged**